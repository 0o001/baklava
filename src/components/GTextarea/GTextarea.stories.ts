import { storiesOf } from '@storybook/vue';
import GTextarea from './GTextarea.vue';

storiesOf('GTextarea', module)
  .addParameters({ component: GTextarea })
  .add('Default', () => ({
    components: { GTextarea },
    data () {
      return {
        email: '',
      };
    },
    template: `
    <div style="margin: 40px 200px 200px">
      <GTextarea class="g-mb-20" label="Email"  v-model="email"/>
      EMAIL: {{email}}<br><br><br><br>
      <GTextarea class="g-mb-20" label="Email" icon="info" />
      <GTextarea class="g-mb-20" label="Email" feedback="feedback" />
      <GTextarea class="g-mb-20" label="Email" feedback="feedback" isOutlineLabel/>
     </div>`,
  }))
  .add('States', () => ({
    components: { GTextarea },
    template: `
    <div style="margin: 40px 200px 200px">
    <GTextarea class="g-mb-20" label="Email" disable  feedback="feedback" isOutlineLabel/>
    <GTextarea class="g-mb-20" label="Email" disable icon="info"/>
    <GTextarea class="g-mb-20" label="Email" error  feedback="feedback" isOutlineLabel/>
    <GTextarea class="g-mb-20" label="Email" error icon="info"/>
    <GTextarea class="g-mb-20" label="Email" success  feedback="feedback" isOutlineLabel/>
    <GTextarea class="g-mb-20" label="Email" success icon="info"/>
    </div>`,
  }));
