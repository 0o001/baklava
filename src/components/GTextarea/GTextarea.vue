<template>
  <GFieldWrapper
    :is-value="!!value"
    :is-outline-label="isOutlineLabel"
    :feedback="feedback"
    :label="label"
    :icon="icon"
    :success="success"
    :error="error"
    :disable="disable"
  >
    <textarea
      :value="getValue"
      @input="onInput"
      v-on="listeners"
      v-bind="$attrs"
    />
  </GFieldWrapper>
</template>

<script lang="ts">

import ClickOutside from '../../directives/ClickOutside.ts';
import GFieldWrapper from '../GFieldWrapper/GFieldWrapper.vue';

export default {
  name: 'GTextarea',
  components: {
    GFieldWrapper,
  },
  directives: {
    ClickOutside,
  },
  props: {
    value: {
      type: String,
      default: () => null,
    },
    type: {
      default: 'text',
      type: String,
    },
    isOutlineLabel: {
      default: false,
      type: Boolean,
    },
    success: {
      default: false,
      type: Boolean,
    },
    error: {
      default: false,
      type: Boolean,
    },
    disable: {
      default: false,
      type: Boolean,
    },
    label: {
      default: '',
      type: String,
    },
    feedback: {
      default: '',
      type: String,
    },
    icon: {
      default: '',
      type: String,
    },
  },

  computed: {
    getValue () {
      return this.value;
    },

    listeners () {
      const { input, ...rest } = this.$listeners;

      return rest;
    },
  },

  methods: {
    onInput (e: any) {
      this.$emit('input', e.target.value);
    },
  },
};
</script>

<style lang="scss" scoped>
  textarea {
    min-height: 32px;
  }
</style>
