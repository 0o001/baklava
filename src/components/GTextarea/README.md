# GTextarea

## Props

| Name               | Type      | Description | Default      |
| ------------------ | --------- | ----------- | ------------ |
| `v-model`          | `String`  |             | `() => null` |
| `type`             | `String`  |             | `"text"`     |
| `is-outline-label` | `Boolean` |             | `false`      |
| `success`          | `Boolean` |             | `false`      |
| `error`            | `Boolean` |             | `false`      |
| `disable`          | `Boolean` |             | `false`      |
| `label`            | `String`  |             | `""`         |
| `feedback`         | `String`  |             | `""`         |
| `icon`             | `String`  |             | `""`         |

## Computed Properties

| Name        | Type      | Description                    |
| ----------- | --------- | ------------------------------ |
| `getValue`  | `object`  | **Dependencies:** `value`      |
| `listeners` | `unknown` | **Dependencies:** `$listeners` |

## Events

| Name    | Description                                               |
| ------- | --------------------------------------------------------- |
| `input` | <br/>**Arguments**<br/><ul><li>**`value: any`**</li></ul> |

## Methods

### onInput()

**Syntax**

```typescript
onInput(e: any): void
```

