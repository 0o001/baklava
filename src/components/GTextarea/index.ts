import GTextarea from './GTextarea.vue';

export default GTextarea;
