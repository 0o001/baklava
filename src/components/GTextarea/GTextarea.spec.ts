import { mount } from '@vue/test-utils';
import GTextarea from '.';

describe('GTextarea', () => {
  let wrapper: any;

  it('should match snapshot when show is prop is given as false ', () => {
    wrapper = mount(GTextarea);

    expect(wrapper.element).toMatchSnapshot();
  });
});