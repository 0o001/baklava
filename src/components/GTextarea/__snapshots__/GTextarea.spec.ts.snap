// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTextarea should match snapshot when show is prop is given as false  1`] = `
<div>
  <div
    class="g-field-wrapper -borderline big"
  >
    <label />
     
    <div
      class="content"
    >
      <textarea />
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;
